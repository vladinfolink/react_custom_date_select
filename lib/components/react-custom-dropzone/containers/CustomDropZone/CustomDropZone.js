"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var React = tslib_1.__importStar(require("react"));
var CustomDropzoneContainer_1 = require("../../../../components/react-custom-dropzone/containers/CustomDropzoneContainer/CustomDropzoneContainer");
var Files_1 = require("../../view/Files/Files");
var RejectedFiles_1 = require("../../view/RejectedFiles/RejectedFiles");
var CustomDropZone = (function (_super) {
    tslib_1.__extends(CustomDropZone, _super);
    function CustomDropZone(props) {
        var _this = _super.call(this, props) || this;
        _this.handleDrop = function (fileItems) {
            var typesAllowed = _this.props.fileTypesAllowed.slice();
            var maxFileSize = _this.props.maxFileSize;
            var _a = _this.state, files = _a.files, rejectedFiles = _a.rejectedFiles;
            var inboundFiles = !!typesAllowed.length ? Array.from(fileItems).reduce(function (ini, item) {
                return (typesAllowed.indexOf(item.type) !== -1 && item.size <= maxFileSize)
                    ? tslib_1.__assign({}, ini, { files: ini.files.concat([item]) }) : tslib_1.__assign({}, ini, { rejectedFiles: ini.rejectedFiles.concat([item]) });
            }, { files: [], rejectedFiles: [] }) : {
                files: Array.from(fileItems), rejectedFiles: []
            };
            for (var _i = 0, _b = inboundFiles.files; _i < _b.length; _i++) {
                var iterator = _b[_i];
                if (files.length < _this.props.fileCountLimit) {
                    files.push(iterator);
                }
            }
            for (var _c = 0, _d = inboundFiles.rejectedFiles; _c < _d.length; _c++) {
                var iterator = _d[_c];
                if (rejectedFiles.length < _this.props.rejectedFileCountLimit) {
                    rejectedFiles.push(iterator);
                }
            }
            return _this.updateState({ files: files, rejectedFiles: rejectedFiles });
        };
        _this.deleteFile = function (fileIndex) {
            var files = _this.state.files.slice();
            files.splice(fileIndex, 1);
            _this.setState(function () {
                return { files: files };
            }, function () {
                _this.props.handleFiles(tslib_1.__assign({}, _this.state));
            });
        };
        _this.state = {
            files: [],
            rejectedFiles: []
        };
        _this.updateState = _this.updateState.bind(_this);
        _this.shouldDisplayFiles = _this.shouldDisplayFiles.bind(_this);
        return _this;
    }
    CustomDropZone.prototype.updateState = function (combinedFilesObj) {
        var _this = this;
        var files = combinedFilesObj.files, rejectedFiles = combinedFilesObj.rejectedFiles;
        this.setState(function () {
            return {
                files: files,
                rejectedFiles: rejectedFiles
            };
        }, function () {
            _this.props.handleFiles(tslib_1.__assign({}, _this.state));
        });
    };
    CustomDropZone.prototype.shouldDisplayFiles = function () {
        return this.props.displayFiles;
    };
    CustomDropZone.prototype.render = function () {
        var dropText = !this.state.files.length ? this.props.dropText : null;
        return (React.createElement(React.Fragment, null,
            React.createElement(CustomDropzoneContainer_1.CustomDropZoneContainer, { dropText: dropText, handleDrop: this.handleDrop, allowClick: this.props.allowClick },
                this.shouldDisplayFiles() && (React.createElement(Files_1.Files, { renderPics: this.props.renderPics, files: this.state.files, deleteFile: this.deleteFile })),
                !!this.state.rejectedFiles.length &&
                    this.props.displayRejectedFiles && (React.createElement(RejectedFiles_1.RejectedFiles, { rejectedFiles: this.state.rejectedFiles })))));
    };
    CustomDropZone.defaultProps = {
        renderPics: true,
        allowClick: true,
        displayFiles: true,
        displayRejectedFiles: true,
        fileCountLimit: Infinity,
        rejectedFileCountLimit: Infinity,
        dropText: 'Drop or select files to upload.',
        handleFiles: function (params) {
            console.log(params);
        },
        fileTypesAllowed: [],
        maxFileSize: Infinity
    };
    return CustomDropZone;
}(React.Component));
exports.CustomDropZone = CustomDropZone;
//# sourceMappingURL=CustomDropZone.js.map