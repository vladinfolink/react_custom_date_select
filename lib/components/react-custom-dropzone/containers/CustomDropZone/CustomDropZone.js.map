{"version":3,"file":"CustomDropZone.js","sourceRoot":"","sources":["../../../../../src/components/react-custom-dropzone/containers/CustomDropZone/CustomDropZone.tsx"],"names":[],"mappings":";;;AACA,mDAA+B;AAE/B,mJAAkJ;AAClJ,gDAA+C;AAC/C,wEAAuE;AAoBvE;IAAoC,0CAA+B;IAiBjE,wBAAY,KAAU;QAAtB,YACE,kBAAM,KAAK,CAAC,SAOb;QAEO,gBAAU,GAAG,UAAC,SAAgB;YACpC,IAAM,YAAY,GAAc,KAAI,CAAC,KAAK,CAAC,gBAAgB,QAAC,CAAC;YAC7D,IAAM,WAAW,GAAW,KAAI,CAAC,KAAK,CAAC,WAAW,CAAC;YAC7C,IAAA,gBAAqC,EAAnC,gBAAK,EAAE,gCAA4B,CAAC;YAE5C,IAAM,YAAY,GAAG,CAAC,CAAC,YAAY,CAAC,MAAM,CAAC,CAAC,CAAC,KAAK,CAAC,IAAI,CAAC,SAAS,CAAC,CAAC,MAAM,CACvE,UAAU,GAAG,EAAE,IAAI;gBACjB,OAAO,CAAC,YAAY,CAAC,OAAO,CAAC,IAAI,CAAC,IAAI,CAAC,KAAK,CAAC,CAAC,IAAI,IAAI,CAAC,IAAI,IAAI,WAAW,CAAC;oBACzE,CAAC,sBAAM,GAAG,IAAE,KAAK,EAAM,GAAG,CAAC,KAAK,SAAE,IAAI,MACtC,CAAC,sBAAM,GAAG,IAAE,aAAa,EAAM,GAAG,CAAC,aAAa,SAAE,IAAI,KAAG,CAAC;YAC9D,CAAC,EACD,EAAE,KAAK,EAAE,EAAE,EAAE,aAAa,EAAE,EAAE,EAAE,CACjC,CAAC,CAAC,CAAC;gBACA,KAAK,EAAE,KAAK,CAAC,IAAI,CAAC,SAAS,CAAC,EAAE,aAAa,EAAE,EAAE;aAChD,CAAC;YAEJ,KAAuB,UAAkB,EAAlB,KAAA,YAAY,CAAC,KAAK,EAAlB,cAAkB,EAAlB,IAAkB,EAAE;gBAAtC,IAAM,QAAQ,SAAA;gBACjB,IAAI,KAAK,CAAC,MAAM,GAAG,KAAI,CAAC,KAAK,CAAC,cAAc,EAAE;oBAC5C,KAAK,CAAC,IAAI,CAAC,QAAQ,CAAC,CAAC;iBACtB;aACF;YAED,KAAuB,UAA0B,EAA1B,KAAA,YAAY,CAAC,aAAa,EAA1B,cAA0B,EAA1B,IAA0B,EAAE;gBAA9C,IAAM,QAAQ,SAAA;gBACjB,IAAI,aAAa,CAAC,MAAM,GAAG,KAAI,CAAC,KAAK,CAAC,sBAAsB,EAAE;oBAC5D,aAAa,CAAC,IAAI,CAAC,QAAQ,CAAC,CAAC;iBAC9B;aACF;YACD,OAAO,KAAI,CAAC,WAAW,CAAC,EAAE,KAAK,OAAA,EAAE,aAAa,eAAA,EAAE,CAAC,CAAC;QACpD,CAAC,CAAC;QAiBM,gBAAU,GAAG,UAAC,SAAiB;YACrC,IAAM,KAAK,GAAO,KAAI,CAAC,KAAK,CAAC,KAAK,QAAC,CAAC;YACpC,KAAK,CAAC,MAAM,CAAC,SAAS,EAAE,CAAC,CAAC,CAAC;YAC3B,KAAI,CAAC,QAAQ,CACX;gBACE,OAAO,EAAE,KAAK,OAAA,EAAE,CAAC;YACnB,CAAC,EACD;gBACE,KAAI,CAAC,KAAK,CAAC,WAAW,sBAAM,KAAI,CAAC,KAAK,EAAG,CAAC;YAC5C,CAAC,CACF,CAAC;QACJ,CAAC,CAAC;QAhEA,KAAI,CAAC,KAAK,GAAG;YACX,KAAK,EAAE,EAAE;YACT,aAAa,EAAE,EAAE;SAClB,CAAC;QACF,KAAI,CAAC,WAAW,GAAG,KAAI,CAAC,WAAW,CAAC,IAAI,CAAC,KAAI,CAAC,CAAC;QAC/C,KAAI,CAAC,kBAAkB,GAAG,KAAI,CAAC,kBAAkB,CAAC,IAAI,CAAC,KAAI,CAAC,CAAC;;IAC/D,CAAC;IAgCO,oCAAW,GAAnB,UAAoB,gBAAoD;QAAxE,iBAaC;QAZS,IAAA,8BAAK,EAAE,8CAAa,CAAsB;QAClD,IAAI,CAAC,QAAQ,CACX;YACE,OAAO;gBACL,KAAK,OAAA;gBACL,aAAa,eAAA;aACd,CAAC;QACJ,CAAC,EACD;YACE,KAAI,CAAC,KAAK,CAAC,WAAW,sBAAM,KAAI,CAAC,KAAK,EAAG,CAAC;QAC5C,CAAC,CACF,CAAC;IACJ,CAAC;IAeO,2CAAkB,GAA1B;QACE,OAAO,IAAI,CAAC,KAAK,CAAC,YAAY,CAAA;IAChC,CAAC;IAED,+BAAM,GAAN;QACE,IAAM,QAAQ,GAAG,CAAC,IAAI,CAAC,KAAK,CAAC,KAAK,CAAC,MAAM,CAAC,CAAC,CAAC,IAAI,CAAC,KAAK,CAAC,QAAQ,CAAC,CAAC,CAAC,IAAI,CAAC;QAEvE,OAAO,CACL,oBAAC,KAAK,CAAC,QAAQ;YACb,oBAAC,iDAAuB,IACtB,QAAQ,EAAE,QAAQ,EAClB,UAAU,EAAE,IAAI,CAAC,UAAU,EAC3B,UAAU,EAAE,IAAI,CAAC,KAAK,CAAC,UAAU;gBAEhC,IAAI,CAAC,kBAAkB,EAAE,IAAI,CAC5B,oBAAC,aAAK,IACJ,UAAU,EAAE,IAAI,CAAC,KAAK,CAAC,UAAU,EACjC,KAAK,EAAE,IAAI,CAAC,KAAK,CAAC,KAAK,EACvB,UAAU,EAAE,IAAI,CAAC,UAAU,GAC3B,CACH;gBAEC,CAAC,CAAC,IAAI,CAAC,KAAK,CAAC,aAAa,CAAC,MAAM;oBACjC,IAAI,CAAC,KAAK,CAAC,oBAAoB,IAAI,CACjC,oBAAC,6BAAa,IAAC,aAAa,EAAE,IAAI,CAAC,KAAK,CAAC,aAAa,GAAI,CAC3D,CACqB,CACX,CAClB,CAAC;IACJ,CAAC;IAhHM,2BAAY,GAAG;QACpB,UAAU,EAAE,IAAI;QAChB,UAAU,EAAE,IAAI;QAChB,YAAY,EAAE,IAAI;QAClB,oBAAoB,EAAE,IAAI;QAC1B,cAAc,EAAE,QAAQ;QACxB,sBAAsB,EAAE,QAAQ;QAChC,QAAQ,EAAE,iCAAiC;QAC3C,WAAW,EAAE,UAAC,MAAW;YACvB,OAAO,CAAC,GAAG,CAAC,MAAM,CAAC,CAAC;QACtB,CAAC;QACD,gBAAgB,EAAE,EAAE;QACpB,WAAW,EAAE,QAAQ;KACtB,CAAA;IAoGH,qBAAC;CAAA,AAnHD,CAAoC,KAAK,CAAC,SAAS,GAmHlD;AAnHY,wCAAc"}