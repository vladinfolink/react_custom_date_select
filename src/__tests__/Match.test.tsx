import * as React from "react";
import * as ReactTestRenderer from "react-test-renderer";
import {CustomDropZone} from '../components/react-custom-dropzone/containers/CustomDropZone/CustomDropZone';
// import {CustomDropZone as NewCustomDropZone} from '../components/react-custom-dropzone/containers/CustomDropZone/CustomDropZone';

const {create} = ReactTestRenderer;

describe("CustomDropZone component match", function ()  {
  expect.assertions(1)
  test("Matches the snapshot", () => {
    const dz = create(<CustomDropZone fileTypesAllowed={['asd']} />);
    expect(dz.toJSON()).toMatchSnapshot();
  });
});

