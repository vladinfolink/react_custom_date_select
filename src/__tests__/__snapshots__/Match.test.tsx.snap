// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomDropZone component match Matches the snapshot 1`] = `
<div
  className="rcd_dropZoneTarget"
  onClick={[Function]}
  onDragEnter={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDrop={[Function]}
>
  Drop or select files to upload.
  <div
    className="reactCustomDropzoneFiles"
  >
    <div
      className="allowedFilesContainer"
    >
       
    </div>
  </div>
  <input
    multiple={true}
    onChange={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    type="file"
  />
</div>
`;
